app/javascript/components/miq-ae-class/index.jsx

```diff
@@ -0,0 +1,193 @@
+import React, { useState, useEffect } from 'react';
+import { FormSpy } from '@data-driven-forms/react-form-renderer';
+import { Button } from 'carbon-components-react';
+import MiqFormRenderer, { useFormApi } from '@@ddf';
+import PropTypes from 'prop-types';
+import createSchema from './class-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import miqFlash from '../../helpers/miq-flash';
+
+const MiqAeClass = ({ classRecord, fqname }) => {
+  const [data, setData] = useState({
+    isLoading: true,
+    initialValues: undefined,
+  });
+
+  const isEdit = !!(classRecord && classRecord.id);
+
+  useEffect(() => {
+    if (isEdit) {
+      http.get(`/miq_ae_class/edit_class_record/${classRecord.id}/`).then((recordValues) => {
+        if (recordValues) {
+          setData({ ...data, isLoading: false, initialValues: recordValues });
+        }
+      });
+    } else {
+      const initialValues = {
+        fqname,
+        name: classRecord && classRecord.name,
+        display_name: classRecord && classRecord.display_name,
+        description: classRecord && classRecord.description,
+      };
+      setData({ ...data, isLoading: false, initialValues });
+    }
+  }, [classRecord]);
+
+  const onSubmit = (values) => {
+    miqSparkleOn();
+
+    const params = {
+      action: isEdit ? 'edit' : 'create',
+      name: values.name,
+      display_name: values.display_name,
+      description: values.description,
+      old_data: data.initialValues,
+      button: classRecord.id ? 'save' : 'add',
+    };
+
+    const request = isEdit
+      ? http.post(`/miq_ae_class/class_update/${classRecord.id}`, params)
+      : http.post(`/miq_ae_class/class_update/`, params);
+
+    request
+      .then((response) => {
+        if (response.status === 200) {
+          const confirmation = isEdit ? __(`Class "${values.name}" was saved`) : __(`Class "${values.name}" was added`);
```

Comment:
For translations in the UI code, you have to use placeholders within the translated text and then fill them in afterwards with something like sprintf. See https://github.com/ManageIQ/guides/blob/master/i18n.md#javascript, particularly the caveats section.  I _think_ this might work:

```suggestion
          let confirmation = isEdit ? __(`Class "%s" was saved`) : __(`Class "%s" was added`);
          confirmation = sprintf(confirmation, value.name);
```

Reply:
Thanks for pointing that out. I've updated that part.